{{cookiecutter.project_slug}}/{{cookiecutter.project_slug}}/users/forms.py

```diff
@@ -12,20 +11,9 @@ class Meta(admin_forms.UserChangeForm.Meta):
 
 
 class UserCreationForm(admin_forms.UserCreationForm):
-
-    error_message = admin_forms.UserCreationForm.error_messages.update(
-        {"duplicate_username": _("This username has already been taken.")}
-    )
-
     class Meta(admin_forms.UserCreationForm.Meta):
         model = User
 
-    def clean_username(self):
-        username = self.cleaned_data["username"]
-
-        try:
-            User.objects.get(username=username)
-        except User.DoesNotExist:
-            return username
-
-        raise ValidationError(self.error_messages["duplicate_username"])
+        error_messages = {
```

Comment:
Does this override the other error messages?

Reply:
Yes @Andrew-Chen-Wang

Reply:
That seems wrong, the base class has another error message that we should keep:

https://github.com/django/django/blob/d8dfff2ab0edf7a1ca5255eccf45c447b2f9d57e/django/contrib/auth/forms.py#L80-L82

Doesn't seem covered by tests btw, if we wrote one we'd get a `KeyError` here:

https://github.com/django/django/blob/d8dfff2ab0edf7a1ca5255eccf45c447b2f9d57e/django/contrib/auth/forms.py#L111

Reply:
@browniebroke My change only overrides the error message in case **unique username validation fails**. Other checks for passwords, emails etc (other fields in the User model would remain as it is). And that is why I override **[error_messages.username.unique](https://github.com/django/django/blob/d8dfff2ab0/django/contrib/auth/models.py#L336)**:

```
        error_messages = {
            "username": {"unique": _("This username has already been taken.")}
        }

```
In fact, in case one adds more checks for the username field (for example every username should start with "@" etc) they should simply get appended to the list of error messages if any. And that is the use-case where overriding the **clean_username() method**  would make more sense, in my opinion.

You are 100% right about the **super().clean_username()**, I am not even sure how I confused RemoteUserBackend with ModelBackend. My bad.

To Summarise the only thing this code would do will change the error message that Django will output by default 
```A user with that username already exists.``` to ```This username has already been taken.``` when unique username validation fails. Nothing else would get affected.

Reply:
Ok I misread the code 🤦  

I thought you were setting this `error_messages` class attribute on the form directly, but I just realised this is on the form nested `Meta` class (as [per documentation](https://docs.djangoproject.com/en/3.0/topics/forms/modelforms/#overriding-the-default-fields)).

This is fine, it just overrides the error message for the specific validation rule, as we want.